src/acl/agentClass.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import {
+  IriString,
+  WithResourceInfo,
+  unstable_WithAcl,
+  unstable_AccessModes,
+  unstable_AclDataset,
+  unstable_AclRule,
+} from "../interfaces";
+import { acl, foaf } from "../constants";
+import { getIriAll } from "../thing/get";
+import {
+  internal_getAclRules,
+  internal_getResourceAclRulesForResource,
+  internal_getDefaultAclRulesForResource,
+  internal_getAccessModes,
+  internal_combineAccessModes,
+  unstable_hasResourceAcl,
+  unstable_hasFallbackAcl,
+} from "../acl";
+
+/**
+ * Find out what Access Modes have been granted to everyone for a given Resource.
+ *
+ * Keep in mind that this function will not tell you what access specific Agents have through other ACL rules, e.g. agent- or group-specific permissions.
+ *
+ * Also, please note that this function is still experimental: its API can change in non-major releases.
+ *
+ * @param resourceInfo Information about the Resource to which the given Agent may have been granted access.
+ * @returns Which Access Modes have been granted to everyone for the given LitDataset, or `null` if it could not be determined (e.g. because the current user does not have Control Access to a given Resource or its Container).
+ */
+export function unstable_getPublicAccessModes(
```

Comment:
I guess it's a more general question of allowing plurals in method names (or class names)...? @ajacksified - not sure if you have an opinion on this general question...?

Reply:
I'll resolve this one - let's keep that discussion at https://github.com/inrupt/lit-pod/pull/212#discussion_r447661698 to keep it at a single place. (Or possible in the coding guidelines; given that it's already in the codebase, we shouldn't really block this PR over this.)